test/k8s_logging_tests/test_config_logging.py

```diff
@@ -38,139 +48,145 @@ def test_cluster_name(setup, test_input, expected):
-    # Data generator image metadata
-    image_name = "cp-data-gen"
-    image_address = "chaitanyaphalak/kafkadatagen:1.0-4-gca7f6d8"
-    image_pull_policy = "IfNotPresent"
-
-    # Create pod in the test namespace to generate logs
-    pod = client.V1Pod()
-    pod.metadata = client.V1ObjectMeta(name=image_name)
-
-    container = client.V1Container(name=image_name, image=image_address, image_pull_policy=image_pull_policy)
+    logger.info("Splunk received %s events in the last minute",
+                len(events))
+    assert len(events) >= expected
 
-    spec = client.V1PodSpec(containers=[container])
-    pod.spec = spec
-    try:
-        v1.create_namespaced_pod(namespace=namespace, body=pod)
-    except ApiException as e:
-        logging.getLogger().info("Exception when calling CoreV1Api create_namespaced_pod: {0}".format(e))
 
-    logging.getLogger().info("Sleeping for 60 seconds")
-    time.sleep(60)
+@pytest.mark.parametrize("container_name,index,expected", [
+    ("pod-w-index-w-ns-index", "pod-anno", 1),
+    ("pod-wo-index-w-ns-index", "ns-anno", 1),
+    ("pod-w-index-wo-ns-index", "pod-anno", 1),
+    ("pod-wo-index-wo-ns-index", os.environ["CI_INDEX_EVENTS"]
+     if os.environ["CI_INDEX_EVENTS"] else "circleci_events", 1),
+])
+def test_annotation_routing(setup, container_name, index, expected):
+    '''
+    Test annotation routing feature. it tests different combinations of 
+    namespace annotations and pod annotations.
+    '''
+    logger.info("testing test_annotation_routing pod={0} index={1} expected={2} event(s)".format(
+        container_name, index, expected))
+    search_query = "index=" + index + " container_name::" + container_name
+    events = check_events_from_splunk(start_time="-1h@h",
+                                      url=setup["splunkd_url"],
+                                      user=setup["splunk_user"],
+                                      query=["search {0}".format(
+                                          search_query)],
+                                      password=setup["splunk_password"])
+    logger.info("Splunk received %s events in the last minute",
+                len(events))
+    assert len(events) >= expected
 
-    # Check if we have those generated logs from kubernetes in Splunk
-    v1.delete_namespaced_pod(name=image_name, namespace=namespace, body=pod)
 
-    search_query = "index={0}".format(test_input)
-    events = check_events_from_splunk(index=index,
-                                      start_time="-1h@h",
+@pytest.mark.parametrize("container_name,expected", [
+    ("pod-w-index-w-ns-exclude", 0),
+    ("pod-w-exclude-wo-ns-exclude", 0)
+])
+def test_annotation_blacklisting(setup, container_name, expected):
+    '''
+    Test annotation blacklisting feature. 
+    '''
+    logger.info("testing test_annotation_blacklisting pod={0} expected={1} event(s)".format(
+        container_name, expected))
+    search_query = "index=*" + " container_name::" + container_name
```

Comment:
this query needs to specifically check for particular indexes instead of "*"

Reply:
in this case, i am testing exclude. So i think it makes sense to leave it as "*" to confirm that it didn't get ingested in ANY indexes.